src/actions/Deselect.ts

```diff
@@ -0,0 +1,41 @@
+import { Selection } from "vscode";
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  TypedSelection,
+  Graph,
+} from "../typings/Types";
+import { runOnTargetsForEachEditor } from "../util/targetUtils";
+
+export default class Deselect implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await runOnTargetsForEachEditor(targets, async (editor, targets) => {
+      // Remove selections with a non-empty intersection
+      const newSelections = editor.selections.filter(
+        (selection) =>
+          !targets.find((target) => {
+            const intersection =
+              target.selection.selection.intersection(selection);
+            return intersection && !intersection.isEmpty;
```

Comment:
Ok I had one more thought: what happens if selection is empty, eg just a cursor? In that case there's no way to deselect it right? Maybe we should allow intersection to be empty if selection was empty to begin with?

Reply:
That is a thought
you could just do a range over it but I get what you mean

Reply:
Implemented and it works quite well. You can reference a token to remove empty selections on both sides of it or you can combine with before and after to just remove on a specific side.